ipa-core/src/protocol/ipa_prf/malicious_security/lagrange.rs

```diff
@@ -35,23 +35,28 @@ where
         // assertion that field is large enough
         // when it is large enough, `F::try_from().unwrap()` below does not panic
         assert!(
-            F::BITS > usize::BITS - N::USIZE.leading_zeros(),
+            u128::from(N::U64) < F::PRIME.into(),
             "Field size {} is not large enough to hold {} points",
-            F::BITS,
-            N::USIZE
+            F::PRIME.into(),
+            N::U64
         );
 
         // assertion that table is not too large for the stack
         assert!(<F as Serializable>::Size::USIZE * N::USIZE < 2024);
 
-        let mut denominator = GenericArray::generate(|_| F::ONE);
-        for (d, i) in denominator.iter_mut().zip(0u64..) {
-            for j in (0..N::U64).filter(|&j| i != j) {
-                *d *= F::try_from(u128::from(i)).unwrap() - F::try_from(u128::from(j)).unwrap();
-            }
-            *d = d.invert();
+        Self {
+            denominator: (0..u128::from(N::U64))
```

Comment:
shouldn't `N::U128` work out of the box w/o extra casts here? https://docs.rs/typenum/latest/typenum/marker_traits/trait.Unsigned.html#associatedconstant.U128

Reply:
You are right. @martinthomson helped me figure out we could get this by bumping the version of `typenum`